api/views/workflow.py

```diff
@@ -35,7 +35,7 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = SourceSerializer
 
     def get_queryset(self):
-        queryset = Source.objects.all()
+        queryset = Source.objects.exclude(icon='')
```

Comment:
File/ImageFields are weird, and I feel like I abused them pretty hard to store icons in the DB. The string would usually be a path on a filesystem...